src/frontend/e2e/tests/upload.spec.ts

```diff
@@ -1,95 +1,105 @@
+    await basePage.clickByTestId("upload-btn");
 
     //accept site cookies if prompted again
-    await acceptSiteCookieTerms(page);
+    await basePage.acceptCookies();
   });
 
   fileExtensions.forEach((extenstion) => {
-    test(`Should upload ${extenstion.toUpperCase()} sample file`, async ({
-      page,
-    }) => {
+    test(`Should upload ${extenstion.toUpperCase()} sample file`, async () => {
       const samples = [];
-      for (let i = 0; i < 3; i++) {
+      for (let i = 0; i < totalLocations; i++) {
         const defaults = { location: locations[i] };
         samples.push(createSampleUploadData(defaults));
       }
       const uploadData = {
         fileExtension: extenstion,
-        samples: samples,
+        samples,
       };
-      await uploadSampleFiles(page, uploadData);
+      await uploadSampleFiles(basePage, uploadData);
 
       //accept site cookies if prompted again
-      await acceptSiteCookieTerms(page);
+      await basePage.acceptCookies();
 
       //continue button
-      await page.locator('a:has-text("Continue")').click();
+      await basePage.clickElement('a:has-text("Continue")');
 
       //accept terms and conditions
-      await page.locator('input[type="checkbox"]').nth(0).click();
-      await page.locator('input[type="checkbox"]').nth(1).click();
+      const acceptUploadToGroupCheckBox = 0;
+      const acceptCzGenepiTermsCheckBox = 1;
+      await basePage.clickCheckBox(acceptUploadToGroupCheckBox);
+      await basePage.clickCheckBox(acceptCzGenepiTermsCheckBox);
 
-      await page.locator("text=Start Upload").click();
+      await basePage.clickByText("Start Upload");
 
       // show confirmation and finish process
-      await expect(page.locator("text=Upload Complete!")).toBeVisible();
-      await page.locator("text=Go to Samples").click();
+      await expect(await basePage.findByText("Upload Complete!")).toBeVisible();
+      await basePage.clickByText("Go to Samples");
     });
   });
 
-  test(`Should validate collection dates`, async ({ page }) => {
+  test(`Should validate collection dates`, async () => {
     const samples = [];
+    const ignoreLocation = true;
     //overwrite collection dates with invalid values
-    for (let i = 0; i < 3; i++) {
+    for (let i = 0; i < totalLocations; i++) {
       const defaults = { location: locations[i] };
       const sample = createSampleUploadData(defaults);
-      sample.collection_date = " ";
+      sample.collection_date = "20-20-20";
       samples.push(sample);
     }
     const uploadData = {
       fileExtension: "txt",
-      samples: samples,
+      samples,
     };
-    await uploadSampleFiles(page, uploadData);
-    expect(await page.locator(`text=${dateErrorMessage}`).count()).toBe(3);
+    await uploadSampleFiles(basePage, uploadData, ignoreLocation);
+    //accept site cookies if prompted again
+    await basePage.acceptCookies();
+
+    expect(await (await basePage.findByText(dateErrorMessage)).count()).toBe(3);
```

Comment:
why `await` on the counting?

Reply:
without wait PL resolves to {} instead of a value